tools/python/runfiles/runfiles.py

```diff
@@ -161,6 +165,51 @@ def EnvVars(self):
     """
     return self._strategy.EnvVars()
 
+  def CurrentRepository(self, frame = 1):
+    """Returns the canonical name of the caller's Bazel repository.
+
+    This function uses the 'inspect' package to obtain the file containing the
+    frame-th caller of this method and returns the canonical name of the Bazel
+    repository containing that file.
```

Comment:
```suggestion
    NOTE: This function inspects the callstack to determine where in the runfiles
    the caller is located to determine which repository it came from. This may
    fail or produce incorrect results depending on who the caller is. Use the
    `frame` argument to control the stack lookup, or pre-resolve the repository
    via other means to avoid this.
```